card/lib/card/auth/current.rb

```diff
@@ -64,14 +64,18 @@ def with auth_data
           auth_data = { current_id: Card.fetch_id(auth_data) }
         end
 
-        tmp_current = current_id
+        tmp_current_id = current_id
         tmp_as_id = as_id
-        @current_id = auth_data[:current_id]
+        tmp_current = @current
+        tmp_as_card = @as_card
+        self.current_id = auth_data[:current_id]
```

Comment:
any reason why we're mixing `self.current =` and `@as_id =` instead of picking one convention or the other?